src/Articulate/Controllers/MetaWeblogController.cs

```diff
@@ -0,0 +1,49 @@
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Articulate.MetaWeblog;
+using Examine;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ViewEngines;
+using Microsoft.Extensions.Logging;
+using Umbraco.Cms.Core.Web;
+using Umbraco.Cms.Web.Common.Controllers;
+using WilderMinds.MetaWeblog;
+
+namespace Articulate.Controllers
+{
+    public class MetaWeblogController : RenderController
+    {
+        private readonly ArticulateMetaWeblogProvider _articulateMetaWeblogProvider;
+        private readonly MetaWeblogService _metaWeblogService;
+
+        public MetaWeblogController(ILogger<RenderController> logger,
+            ICompositeViewEngine compositeViewEngine,
+            IUmbracoContextAccessor umbracoContextAccessor,
+            ArticulateMetaWeblogProvider articulateMetaWeblogProvider,
```

Comment:
Shouldn't IMetaWeblogProvider be injected here? ArticulateMetaWeblogProvider is the implementation

Reply:
Resolving this comment since this has been implemented/fixed in a different way. We now manually create the ArticulateMetaWeblogProvider based on the current node and then manually create the MetaWeblogService to process the xmlrpc call.